velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -114,9 +114,35 @@ TEST_F(DecimalArithmeticTest, add) {
            {1, 2, std::nullopt, 6, std::nullopt}, DECIMAL(10, 3)),
        makeNullableShortDecimalFlatVector(
            {1, 2, 5, std::nullopt, std::nullopt}, DECIMAL(10, 3))});
+
+  // Addition overflow.
+  VELOX_ASSERT_THROW(
+      testDecimalExpr<TypeKind::LONG_DECIMAL>(
+          {},
+          "c0 + cast(1.00 as decimal(2,0))",
+          {makeLongDecimalFlatVector(
+              {UnscaledLongDecimal::max().unscaledValue()}, DECIMAL(38, 0))}),
+      "Decimal overflow: 99999999999999999999999999999999999999 + 1");
+
+  // Rescaling LHS overflows.
+  VELOX_ASSERT_THROW(
+      testDecimalExpr<TypeKind::LONG_DECIMAL>(
+          {},
+          "c0 + 0.01",
+          {makeLongDecimalFlatVector(
+              {UnscaledLongDecimal::max().unscaledValue()}, DECIMAL(38, 0))}),
+      "Decimal overflow: 99999999999999999999999999999999999999 * 100");
```

Comment:
Isn't it confusing that plus operation throws as error that talks about multiply?

Reply:
@mbasmanova We thought about this. Since decimal addition includes rescale(multiply), this should be expected. We can clarify this in the doc.
The alternative is to directly use `__builtin_add_overflow, __builtin_mul_overflow` here and only say `Decimal overflow` as in Presto Java.
What do you suggest?

Reply:
I think users will be confused by this error message and probably won't read the documentation that explains why this message can occur. Using "Decimal overflow" error message without details is fine. And improvement would be to specify the inputs and the operator, but the operation would need to be + for addition, not *. One way to achieve that would be to pass error message (or part of it) to checkedXxx.

Reply:
If we use `__builtin_add_overflow, __builtin_mul_overflow` directly instead of `checkedXxx`, we can customize the error message. I feel passing an error message to `checkXxx` is awkward from an API perspective.

Reply:
@mbasmanova, I switched to using builtin calls. Please take a look.